[Snyk] Fix for 1 vulnerabilities

mod opened the pull request:
![snyk-top-banner](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### Snyk has created this PR to fix 1 vulnerabilities in the rubygems dependencies of this project.

#### Snyk changed the following file(s):

- `Gemfile`



<details>
<summary>⚠️ <b>Warning</b></summary>

```
Failed to update the Gemfile.lock, please update manually before merging.
```

</details>



#### Vulnerabilities that will be fixed with an upgrade:

|  | Issue | Score | 
:-------------------------:|:-------------------------|:-------------------------
![low severity](https://res.cloudinary.com/snyk/image/upload/w_20,h_20/v1561977819/icon/l.png 'low severity') | Buffer Under-read <br/>[SNYK-RUBY-NOKOGIRI-9789079](https://snyk.io/vuln/SNYK-RUBY-NOKOGIRI-9789079) | &nbsp;&nbsp;**498**&nbsp;&nbsp; 




---

> [!IMPORTANT]
>
> - Check the changes in this PR to ensure they won't cause issues with your project.
> - Max score is 1000. Note that the real score may have changed since the PR was raised.
> - This PR was automatically created by Snyk using the credentials of a real user.

---

**Note:** _You are seeing this because you or someone else with access to this repository has authorized Snyk to open fix PRs._

For more information: <img src="https://api.segment.io/v1/pixel/track?data=eyJ3cml0ZUtleSI6InJyWmxZcEdHY2RyTHZsb0lYd0dUcVg4WkFRTnNCOUEwIiwiYW5vbnltb3VzSWQiOiIyYTZhOGE5NS0wZTU2LTRhYzItYjA1Yy0yYzE3MDA4OWNkYTMiLCJldmVudCI6IlBSIHZpZXdlZCIsInByb3BlcnRpZXMiOnsicHJJZCI6IjJhNmE4YTk1LTBlNTYtNGFjMi1iMDVjLTJjMTcwMDg5Y2RhMyJ9fQ==" width="0" height="0"/>
🧐 [View latest project report](https://app.snyk.io/org/mod/project/478f2032-55a5-4895-864a-35b37f77eaa9?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr)
📜 [Customise PR templates](https://docs.snyk.io/scan-using-snyk/pull-requests/snyk-fix-pull-or-merge-requests/customize-pr-templates?utm_source=github&utm_content=fix-pr-template)
🛠 [Adjust project settings](https://app.snyk.io/org/mod/project/478f2032-55a5-4895-864a-35b37f77eaa9?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr/settings)
📚 [Read about Snyk's upgrade logic](https://docs.snyk.io/scan-with-snyk/snyk-open-source/manage-vulnerabilities/upgrade-package-versions-to-fix-vulnerabilities?utm_source=github&utm_content=fix-pr-template)

---

**Learn how to fix vulnerabilities with free interactive lessons:**

🦉 [Learn about vulnerability in an interactive lesson of Snyk Learn.](https://learn.snyk.io/?loc&#x3D;fix-pr)

[//]: # 'snyk:metadata:{"customTemplate":{"variablesUsed":[],"fieldsUsed":[]},"dependencies":[{"name":"rails","from":"5.2.4.4","to":"7.0.0"},{"name":"vault-rails","from":"0.6.0","to":"0.7.1"}],"env":"prod","issuesToFix":["SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079","SNYK-RUBY-NOKOGIRI-9789079"],"prId":"2a6a8a95-0e56-4ac2-b05c-2c170089cda3","prPublicId":"2a6a8a95-0e56-4ac2-b05c-2c170089cda3","packageManager":"rubygems","priorityScoreList":[498],"projectPublicId":"478f2032-55a5-4895-864a-35b37f77eaa9","projectUrl":"https://app.snyk.io/org/mod/project/478f2032-55a5-4895-864a-35b37f77eaa9?utm_source=github&utm_medium=referral&page=fix-pr","prType":"fix","templateFieldSources":{"branchName":"default","commitMessage":"default","description":"default","title":"default"},"templateVariants":["pr-warning-shown","priorityScore"],"type":"auto","upgrade":["SNYK-RUBY-NOKOGIRI-9789079"],"vulns":["SNYK-RUBY-NOKOGIRI-9789079"],"patch":[],"isBreakingChange":true,"remediationStrategy":"vuln"}'

yivo commented:
@mod 

Finished in 3 minutes 37.7 seconds (files took 7.57 seconds to load)
581 examples, 0 failures, 10 pending

yivo commented on app/api/api_v2/auth/jwt_authenticator.rb:
```diff
@@ -0,0 +1,33 @@
+
+      def fetch_member(payload)
+        return nil if payload['member_id'].blank?
+        Member.find_by_id(payload['member_id'])
```
@mod ??? This is old code. How did you get to it? The new code looks like 
<img width="593" alt="screen shot 2017-12-08 at 1 32 43 am" src="https://user-images.githubusercontent.com/7421323/33743967-b895d6d6-dbb7-11e7-889c-45f8be18901c.png">

snyk-bot committed:
fix: Gemfile to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-RUBY-NOKOGIRI-9789079
